pkg/client/cache/expiration_cache_test.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestTTLExpirationBasic(t *testing.T) {
```

Comment:
My issue with this approach is that, while you effectively test the expiration mechanism, you don't test that the expiration happens on the correct condition (time elapsed). So you need to at least test TTLPolicy with a fake clock (see the util package, I think it has one).